### PR TITLE
Rover: Add mis-done-behave parameter.

### PR DESCRIPTION
**Pull request by tajisoft**
@rmackay9 Hello, I'm trying to add mis-done-behave for #9328.

mis-done-behave has two values.
0: HOLD
1: LOITER - only boat can become. Do you think this is a good idea?
Should I also have guided (2: GUIDED) for the rover?
I'm happy to get advice.

### CODE REVIEW

File: APMrover2/Rover.h [hunk 1]
```diff
@@ -265,6 +265,10 @@ class Rover : public AP_HAL::HAL::Callbacks {
         uint32_t start_time;
         uint8_t triggered;
         uint32_t last_valid_rc_ms;
+
+	uint8_t pilot_input          : 1; // true if pilot input failsafe is active, handles things like joystick being disconnected during operation
```
Comment (muramura) [inline APMrover2/Rover.h]: I think it would be better to indent it.

File: APMrover2/mode.cpp [hunk 2]
```diff
@@ -23,21 +23,16 @@ bool Mode::enter()
     const bool ignore_checks = !hal.util->get_soft_armed();   // allow switching to any mode if disarmed.  We rely on the arming check to perform
     if (!ignore_checks) {
 
-        // get EKF filter status
+	// get EKF filter status
```
Comment (muramura) [inline APMrover2/mode.cpp]: I think it would be better to indent it.

File: APMrover2/mode_auto.cpp [hunk 3]
```diff
@@ -75,6 +76,10 @@ void ModeAuto::update()
             break;
         }
 
+	case Auto_Loiter:
```
Comment (muramura) [inline APMrover2/mode_auto.cpp]: I think it would be better to indent it.

File: APMrover2/system.cpp [hunk 4]
```diff
@@ -369,3 +369,21 @@ bool Rover::is_boat() const
 {
     return ((enum frame_class)g2.frame_class.get() == FRAME_BOAT);
 }
+
+// position_ok - returns true if the horizontal absolute position is ok and home position is set
+bool Rover::position_ok(nav_filter_status& filt_status)
```
Comment (muramura) [inline APMrover2/system.cpp]: I think the method name is "is_position_ok".
return status , ok is true, fail is false

File: APMrover2/Rover.cpp [hunk 5]
```diff
@@ -34,6 +34,6 @@ Rover::Rover(void) :
     control_mode(&mode_initializing),
     home(ahrs.get_home()),
     G_Dt(0.02f),
-    mode_auto(mode_rtl)
+    mode_auto(mode_loiter, mode_rtl)
```
Comment (muramura) [inline APMrover2/Rover.cpp]: I set the default value when I generated the constructor.
I think that only one mode will be set.

**Comment by rmackay9** [comment]
@tajisoft, txs for this.  I've reviewed it a bit more and here's some feedback:

1. no need to add mis_done_behave to the Parameters.h enum when the param is added to g2.  To be honest, I'm not sure why but with G2 we magically know it's order.
2. adding loiter to auto is definitely useful but it's not necessary to resolve this issue.  It would be better to simply change the vehicle into Loiter mode if MIS_DONE_BEHAVE = 1.
3. Adding Loiter-in-Auto is great but I think it should be done as a separate PR.  We should use Loiter-in-Auto if the user creates a mission with LOITER_TIME or LOITER_UNLIMITED in it I think.  I don't think loiter_start needs to check if the position is ok.  Instead Rover needs an EKF failsafe.
4. no need to add the get_mis_done_behave() accessor I think.  just directly access the parameter

**Comment by tajisoft** [comment]
@rmackay9 I re-raised PR(#9731).

**Pull request closed without merging**